### PR TITLE
ci(docker): Add workflow to build and publish images

### PR DESCRIPTION
**Pull request by GImbrailo**
The PR sets up the docker build and publish workflow for a list of plugins in a matrix. This is only needed for the `flextesa` plugin currently

**Test Plan**
Please see the Release section of the repo for listed packages

image

image

Please see this PR for the docker comments
 
Ref #38

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+jobs:
+
+  build-and-push-docker:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - plugin: taqueria-plugin-flextesa
```
Comment (GImbrailo) [inline .github/workflows/docker-publish.yml:26]: I think it is useful to run this job as a matrix. This way we can add any new plugins that require docker builds to the matrix. I am happy to remove it for now if needed

File: taqueria-plugin-flextesa/docker/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 FROM oxheadalpha/flextesa:latest
 RUN mkdir /app
 WORKDIR /app
-COPY startFlextesa.js /app/
+COPY startFlextesa.ts /app/
 RUN apk add net-tools && apk add nodejs && apk add npm
```
Comment (jevonearth) [inline taqueria-plugin-flextesa/docker/Dockerfile:5]: Running `RUN apk add net-tools nodejs npm` will complete faster.

**Review by jevonearth — APPROVED** [review]

**Review by jevonearth — APPROVED** [review]

File: .github/workflows/docker-publish.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+    paths:
+      - 'taqueria-plugin-*/docker/*'
+  create:
```
Comment (danielelisi) [inline .github/workflows/docker-publish.yml:12]: I think we should use `push: tags` here like the other workflow

File: .github/workflows/docker-publish.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    tags:
+      - 'v*'
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+
+  build-and-push-docker:
```
Comment (danielelisi) [inline .github/workflows/docker-publish.yml:20]: You can remove the docker suffix since this job is already in the docker workflow

**Review by danielelisi — COMMENTED** [review]
How does this pipeline detect when to push a semver tag? From what I read seems like anyone could push a semver tags from a pr?

File: .github/workflows/docker-publish.yml (hunk 3 above)
Comment (GImbrailo) [reply .github/workflows/docker-publish.yml]: :upside_down_face: Changed

File: .github/workflows/docker-publish.yml (hunk 4 above)
Comment (GImbrailo) [reply .github/workflows/docker-publish.yml]: That makes sense. Changed

**Comment by GImbrailo** [comment]
> How does this pipeline detect when to push a semver tag? From what I read seems like anyone could push a semver tags from a pr?

@danielelisi from my understanding this is done in the `Exctract metadata` job. The tags are set and defined on lines 45 -49 of the workflow

```yaml
          tags: |
            type=ref,event=branch
            type=ref,event=pr,suffix=-{{sha}}
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
```

The documentation on the [docker/metadata-action](https://github.com/docker/metadata-action) repo provide more information on the various types and how to configure them

**Comment by GImbrailo** [comment]
@danielelisi based on the comments in the SRE standup, I am going to make some more changes here

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (GImbrailo) [reply .github/workflows/docker-publish.yml]: As discussed in the SRE standup the `flextesa` docker image is an outlier. I will remove the matrix configuration

**Review by danielelisi — APPROVED** [review]

**Pull request merged**